src/vs/workbench/api/browser/mainThreadTerminalService.ts

```diff
@@ -323,6 +323,13 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 		this._getTerminalProcess(terminalId).then(e => e.emitCwd(cwd));
 	}
 
+	public $sendResolvedLaunchConfig(terminalId: number, shellLaunchConfig: IShellLaunchConfig): void {
+		const instance = this._terminalService.getInstanceFromId(terminalId);
+		if (instance) {
+			instance.shellLaunchConfig = shellLaunchConfig;
```

Comment:
This is a little unsafe allowing anyone to set `shellLaunchConfig`, let's instead set it via `_terminalProcesses`. Check out how `ITerminalProcessExtHostProxy.emitOverrideDimensions` is working, and bubble it up like that. Note that `ITerminalChildProcess.onProcessOverrideDimensions` is optional so it only needs to exist for the ext host proxy, not for regular processes.